collector/src/main/java/io/prometheus/jmx/JmxCollector.java

```diff
@@ -240,38 +245,70 @@ public void recordBean(
           String attrDescription,
           Object beanValue) {
 
-        Number value;
-        if (beanValue instanceof Number) {
-          value = (Number)beanValue;
-        } else if (beanValue instanceof Boolean) {
-          value = (Boolean)beanValue ? 1 : 0;
-        } else {
-          LOGGER.fine("Ignoring non-Number/Boolean bean: " + domain + beanProperties.toString() + attrKeys.toString() + attrName + ": " + beanValue);
-          return;
-        }
-
-        String beanName = domain +
-                   angleBrackets(beanProperties.toString()) +
-                   angleBrackets(attrKeys.toString());
+        String beanName = domain + angleBrackets(beanProperties.toString()) + angleBrackets(attrKeys.toString());
         // attrDescription tends not to be useful, so give the fully qualified name too.
         String help = attrDescription + " (" + beanName + attrName + ")";
-
         String attrNameSnakeCase = snakeCasePattern.matcher(attrName).replaceAll("$1_$2").toLowerCase();
 
         for (Rule rule : rules) {
           Matcher matcher = null;
           String matchName = beanName + (rule.attrNameSnakeCase ? attrNameSnakeCase : attrName);
           if (rule.pattern != null) {
-            matcher = rule.pattern.matcher(matchName + ": " + value);
+            matcher = rule.pattern.matcher(matchName + ": " + beanValue);
             if (!matcher.matches()) {
               continue;
             }
           }
+
+          // Transform/cast value
+          Number value = null;
+          if (rule.value != null && !rule.value.isEmpty()) {
+            String val = matcher.replaceAll(rule.value);
+
+            Matcher math = mathPattern.matcher(val);
```

Comment:
I'd have `value` as a subsitiutable float, and `valueFactor` as number it's multiplied by. The use case is converting milliseconds to seconds, there's no need for arithmetic operations beyond that.